### PR TITLE
issue#479 of the functionality of generate-dashboard button resolved

### PR DESCRIPTION
**Pull request by shradha-khapra**

### CODE REVIEW

File: dashboard-generator/generator.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 // utility functions
 var $ = (selector) => document.querySelector(selector);
-
+var countDash=0; //a variable to keep count of the added dashboards
```
Comment (chutten) [inline dashboard-generator/generator.js]: The style in this file is to put spaces around the equals sign in assignments, so it would be appropriate to include them here, too.

File: dashboard-generator/generator.js [hunk 2]
```diff
@@ -134,14 +135,18 @@ window.addEventListener('load', function () {
       _dash.splice(plotIndex, 1);
       updatePostData();
       tr.parentElement.removeChild(tr);
+      countDash--;
```
Comment (chutten) [inline dashboard-generator/generator.js]: Instead of counting the dashboards independently, could you check the length of `_dash`? That will save us the global counter, and tie our conditions directly to the data model.

File: dashboard-generator/generator.js [hunk 3]
```diff
@@ -134,14 +135,18 @@ window.addEventListener('load', function () {
...
+      if(countDash==0)
```
Comment (chutten) [inline dashboard-generator/generator.js]: In this file spaces are placed between `if` and `(` as well as around `==`

File: dashboard-generator/generator.js [hunk 4]
```diff
@@ -134,14 +135,18 @@ window.addEventListener('load', function () {
...
+        $('#generate').setAttribute('disabled',true);
```
Comment (chutten) [inline dashboard-generator/generator.js]: In this file `if` statements are all at least three lines long, requiring `{}` around their bodies.

File: dashboard-generator/generator.js [hunk 5]
```diff
@@ -134,14 +135,18 @@ window.addEventListener('load', function () {
...
     });
     rmTd.appendChild(rmButton);
     tr.appendChild(rmTd);
 
     $('.dashboard-plots-body').appendChild(tr);
 
     // now that the dash spec has a plot, user can generate a dash
-    $('#generate').removeAttribute('disabled');
+    if(countDash)
+      $('#generate').removeAttribute('disabled');
```
Comment (chutten) [inline dashboard-generator/generator.js]: The same style considerations are required here.

**Review by chutten — CHANGES_REQUESTED** [review]
Thank you for your patch. The change looks like it'll work swimmingly. It needs some changes so the style of the file is uniform throughout, and then we'll be able to get this submitted.

**Comment by shradha-khapra** [comment]
Thank you for suggesting all the changes. I have worked upon all the requested changes and I'll be sending another PR.

**Comment by shradha-khapra** [comment]
I sent another PR with new changes . Please help me out if I missed out on something.

**Comment by chutten** [comment]
I don't see it. Can you provide a link?

**Comment by shradha-khapra** [comment]
I made the changes and edited the existing pull request.

File: dashboard-generator/generator.js [hunk 6]
```diff
@@ -110,6 +110,7 @@ window.addEventListener('load', function () {
   }
 
   function addPlotToTable(plotParams) {
+
```
Comment (chutten) [inline dashboard-generator/generator.js:113]: This whitespace is unrelated to this patch.

File: dashboard-generator/generator.js [hunk 7]
```diff
@@ -134,14 +135,20 @@ window.addEventListener('load', function () {
       _dash.splice(plotIndex, 1);
       updatePostData();
       tr.parentElement.removeChild(tr);
+
+      if ( _dash.length == 0 ) {
```
Comment (chutten) [inline dashboard-generator/generator.js:139]: Style in this file is to not have spaces directly within the parentheses (so this would be `if (_dash.length == 0) {`)

File: dashboard-generator/generator.js [hunk 8]
```diff
@@ -134,14 +135,20 @@ window.addEventListener('load', function () {
...
+      }
     });
     rmTd.appendChild(rmButton);
     tr.appendChild(rmTd);
 
     $('.dashboard-plots-body').appendChild(tr);
 
     // now that the dash spec has a plot, user can generate a dash
-    $('#generate').removeAttribute('disabled');
+    if (_dash.length != 0 ) {
```
Comment (chutten) [inline dashboard-generator/generator.js:149]: Same whitespace consideration here.

**Review by chutten — COMMENTED** [review]
Just some minor whitespace stuff and we should be good to go.
